packages/legacy/utils/src/logger.ts

```diff
@@ -79,18 +79,23 @@ export class DefaultLogger implements Logger {
     const message = this.getLoggerMessage({
       args,
     });
-    return console.log(`${this.prefix} ${message}`);
+    const fullMessage = `${this.prefix} ${message}`;
+    if (typeof process === 'object') {
+      process.stderr.write(fullMessage);
+    } else {
+      console.log(fullMessage);
+    }
   }
 
   warn(...args: any[]) {
     const message = this.getLoggerMessage({
       args,
     });
     const fullMessage = `${this.prefix} ⚠️ ${warnColor(message)}`;
-    if (console.warn) {
-      console.warn(fullMessage);
+    if (typeof process === 'object') {
```

Comment:
This means we are changing the priority of the logging function ? We call `console.warn` only on non-node env where `process` is not available ?

Reply:
What do you mean by changing priority? By convention, stderr is for diagnostics while stdout is for program output. This makes sure various shell techniques still work. Things like piping to a variable or file while still seeing logs (without them being in the destination).